info.go

```diff
@@ -556,6 +566,10 @@ func (pi *ProgramInfo) btfSpec() (*btf.Spec, error) {
 	return spec, nil
 }
 
+// ErrInfoRestricted when kernel.kptr_restrict and net.core.bpf_jit_harden are set, the kernel may restrict access
+// to certain bits of the program info.
+var ErrInfoRestricted = errors.New("program info restricted by combination of kernel.kptr_restrict and net.core.bpf_jit_harden")
```

Comment:
Gut feel: we may need this in more places down the line, and we should do a bit of exploration in the kernel source to figure out which other APIs are affected by kptr_restrict and bpf_jit_harden. (kallsyms comes to mind)

I like the name of the sentinel, but the string is slightly too specific IMO; the context around program info being restricted should be wrapped into the error string by the methods on ProgramInfo.